fe/fe-core/src/main/java/org/apache/doris/statistics/TableStats.java

```diff
@@ -78,7 +78,7 @@ public TableStats(double rowCount, long dataSize) {
     }
 
     public double getRowCount() {
-        if  (rowCount == -1) {
+        if  (rowCount == -1 && !nameToPartitionStats.isEmpty()) {
```

Comment:
Is latter condition necessary?

Reply:
This condition is added mainly because the result returns 0 instead of the expected -1 when `nameToPartitionStats` is empty.

Reply:
it is better add comment to record the reason

Reply:
Sure